### PR TITLE
NonViewComponent attribute documentation

### LINKED ISSUE #23963: Document [NonViewComponent]
In answering a [Stack Overflow question](https://stackoverflow.com/questions/70009195/remove-the-naming-convention-rule-for-finding-viewcomponents/70009443), I just discovered there's a `[NonViewComponent]` attribute for marking a class that ends with `ViewComponent` (case-insensitive) as, well, not being a View Component.

I think we should document this, under https://docs.microsoft.com/en-us/aspnet/core/mvc/views/view-components?view=aspnetcore-6.0#the-view-component-class.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9d2902f4-b6fd-ba1c-6762-d7c26f5eb4b9
* Version Independent ID: b5715d72-5f10-bb93-1051-7f06dfa9cde8
* Content: [View components in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/mvc/views/view-components?view=aspnetcore-6.0#the-view-component-class)
* Content Source: [aspnetcore/mvc/views/view-components.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/mvc/views/view-components.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-mvc**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

### PR DESCRIPTION
**Pull request by Maniizzle**
Fixes #23963

This is a documentation for the use of the NonViewComponent Attribute([NonViewComponent]). 

[Internal Preview URL](https://review.docs.microsoft.com/en-us/aspnet/core/mvc/views/view-components?view=aspnetcore-6.0&branch=pr-en-us-24424#the-view-component-class).

### CODE REVIEW

**Comment by guardrex** [comment]
Hi @Maniizzle ... Kirk is out for the seasonal holiday but will return soon. I know that you're working with him on this, so I'll assign this to him. He'll get back to you when he returns in '22.

**Comment by Maniizzle** [comment]
Thank you @guardrex

**Comment by guardrex** [comment]
Sure thing .... and the build should start working again soon. You can see ☝️ that we have some 😈 **_GREMLINS_** 😈 at the moment *having some holiday fun with us here!* 😆 ... They'll get those little guys out at some point, then this will build out and show you if there are any real build warnings/errors.

**Comment by Maniizzle** [comment]
hahahaha.. I saw the  😈 GREMLINS 😈. They are really having fun

**Comment by guardrex** [comment]
@Rick-Anderson ... I've been periodically trying that on my patch PR, too. No 🎲🎲 yet.

**Comment by Rick-Anderson** [comment]
I'm monitoring the internal bug. I'll keep you posted.

File: aspnetcore/mvc/views/view-components.md [hunk 1]
```diff
@@ -59,6 +59,18 @@ A view component class:
...
+Suppose you have a class in your project with the suffix *ViewComponent*(case-insensitive) and you don't want it to serve as a ViewComponent, you can decorate the class with the attribute `[NonViewComponent]`:
+ 
+```csharp
+     [NonViewComponent]
+     public class ReviewComponent
+     {
+        // .....
+     
+ ```
+
```
Comment (serpent5) [inline aspnetcore/mvc/views/view-components.md:73]: I think this description is short enough that it doesn't need to be its own section. We also generally avoid "you" where we can, so this suggestion does that and includes a cross-link to the attribute:

```suggestion
To stop a class that has a case-insensitive *ViewComponent* suffix from being treated as a view component, decorate the class with the [[NonViewComponent]](xref:Microsoft.AspNetCore.Mvc.NonViewComponentAttribute) attribute:
 
```csharp
[NonViewComponent]
public class ReviewComponent
{
    // ...
```
```

**Review by serpent5 — COMMENTED** [review]
Thank you for doing this, @Maniizzle. I've made a suggestion with a couple of changes around style, but this looks really good and makes for a useful addition.

File: aspnetcore/mvc/views/view-components.md (hunk 1 above)
Comment (Maniizzle) [reply aspnetcore/mvc/views/view-components.md:73]: Thank you @serpent5 for the suggestion. I liked the way it was rephrased. it is succinct. I would go ahead and commit the changes.

Comment (serpent5) [reply aspnetcore/mvc/views/view-components.md]: It looks like my suggestion didn't include the closing backtick, so I've just added that in a commit. I'll review the final result and then merge this when it looks correct.

Thanks again, @Maniizzle. ⭐

**Comment by serpent5** [comment]
@Maniizzle Be sure to check [What's New](https://docs.microsoft.com/en-us/aspnet/core/whats-new/?view=aspnetcore-6.0) early in February, where you'll see yourself listed as a valued contributor for January.

File: aspnetcore/mvc/views/view-components.md (hunk 1 above)
Comment (Maniizzle) [reply aspnetcore/mvc/views/view-components.md]: Learnt few things about documentation from this engagement . Thanks @serpent5

**Pull request merged**